Fix race condition with Ninja >= 1.12.0 where any_rep_types.h is not generated
before Any.cpp is compiled.

https://github.com/facebook/fbthrift/pull/599

--- thrift/lib/cpp2/CMakeLists.txt.orig	2024-05-02 22:42:02 UTC
+++ thrift/lib/cpp2/CMakeLists.txt
@@ -109,8 +109,8 @@ add_library(
   type/TypeRegistry.cpp
   type/UniversalName.cpp
 )
-add_dependencies(thrifttype type-cpp2-target thriftannotation thrifttyperep
-  thriftprotocol)
+add_dependencies(thrifttype any_rep-cpp2-target type-cpp2-target
+  thriftannotation thrifttyperep thriftprotocol)
 target_link_libraries(
   thrifttype
   PUBLIC
